src/update.ts

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+import * as uuid from 'uuid';
+import { bufferToHex } from 'ethereumjs-util'
+import { helpers } from 'leap-core';
+import Web3 from 'web3';
+
+import dynamoDb from './dynamodb'
+
+export const updateStats = async (_event, _context, callback) => {
+  const plasmaProvider = process.env.PLASMA_PROVIDER;
+  console.log('Updating stats... ' + plasmaProvider);
+
+  const web3Plasma = helpers.extendWeb3(new Web3(plasmaProvider));
+
+  await web3Plasma.eth.net.getId();
+
+  const uTxos = await web3Plasma.getUnspent();
+
+  let unspents: import('./types').UnspentWithTx[] = await Promise.all(
+    uTxos.map(async (u: import('./types').UnspentWithTx) => {
+      u.transaction = await web3Plasma.eth.getTransaction(bufferToHex(u.outpoint.hash)) as import('./types').LeapTransaction;
+      return u;
+    })
+  );
+
+  const blockNumbers = [...new Set(unspents.map(u => u.transaction.blockNumber))]; // Removes duplicates
+  const blocksWithTimestamps = {};
+
+  await Promise.all(
+    blockNumbers.map(async (bn) => {
+      const block = await web3Plasma.eth.getBlock(bn);
+      blocksWithTimestamps[bn] = block.timestamp;
+    })
+  );
+
+  unspents = unspents.map(u => {
+    u.timestamp = blocksWithTimestamps[u.transaction.blockNumber];
+    return u;
+  });
+
+  // Filter out last 30 days UTXOs
+  const endTimestamp = new Date().getTime() / 1000;
+  const startTimestamp = endTimestamp - 30 * 24 * 60 * 60; // 30 days old
+
+  const currentUtxos = unspents.filter(u => u.timestamp >= startTimestamp && u.timestamp <= endTimestamp);
+
+  // Remove same address transactions
+  const activeUtxos = currentUtxos.filter(c => c.transaction.from !== c.transaction.to);
```

Comment:
Keep in mind, this works only for "simple" transfer txs. It considers that destination of tx is the very first output, which is not always correct — transaction may split the input between different beneficiaries, spending condition is tricky etc

Similar logic applies to `from` field (see same links)

Reference:
[1] - How we set `to` for transaction we return from JSON RPC in leap-node https://github.com/leapdao/leap-node/blob/23f807ada23c53eb5d7cb2d371e960ef8995a3fe/src/api/methods/txResponse.js#L23
[2] - the code which actually derives `to` attribute: https://github.com/leapdao/leap-core/blob/eec71e1451100fcf2ddaa8f5a7a5722d7fb6bd41/lib/transaction.js#L434

Reply:
just a note. I don't know how to reliably solve this, so it is the best we can have (easily)